### PR TITLE
chore(arm): update error output

### PR DESCRIPTION
**Pull request by xzf0587**
1. remove duplicate error output when arm deployment failed.
2. update error message in notification.
3. add module error property in telemetry.

### CODE REVIEW

**Comment by xzf0587** [comment]
image

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 1]
```diff
@@ -399,10 +401,15 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
         [SolutionTelemetryProperty.Success]: SolutionTelemetrySuccess.Yes,
       });
     } else {
+      const properties: { [key: string]: string } = {};
+      if (result.error.innerError) {
+        properties[SolutionTelemetryProperty.ModuleErrorMessage] = result.error.innerError;
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:406]: can we name the property to `arm-deployment-error` or something else to make it more straightforward?

File: packages/fx-core/src/plugins/solution/fx-solution/arm.ts [hunk 2]
```diff
@@ -399,10 +401,15 @@ export async function deployArmTemplates(ctx: SolutionContext): Promise<Result<v
         [SolutionTelemetryProperty.Success]: SolutionTelemetrySuccess.Yes,
       });
     } else {
+      const properties: { [key: string]: string } = {};
```
Comment (blackchoey) [inline packages/fx-core/src/plugins/solution/fx-solution/arm.ts:404]: better naming for the variable? maybe `errorProperties` or something else

File: packages/vscode-extension/src/handlers.ts [hunk 3]
```diff
@@ -1626,7 +1626,10 @@ export async function showError(e: UserError | SystemError) {
       },
     };
 
-    const button = await window.showErrorMessage(`[${errorCode}]: ${e.message}`, help);
+    const button = await window.showErrorMessage(
+      `[${errorCode}]: ${e.userData ?? e.message}`,
```
Comment (blackchoey) [inline packages/vscode-extension/src/handlers.ts:1630]: Is there any other code using `userData`? Per the comment of userData, seems it's for other purpose

**Pull request closed without merging**